pkg/scheduler/algorithm/predicates/predicates.go

```diff
@@ -830,6 +830,10 @@ func GetResourceRequest(pod *v1.Pod) *schedulernodeinfo.Resource {
 		result.SetMaxResource(container.Resources.Requests)
 	}
 
+	for _, container := range pod.Spec.EphemeralContainers {
```

Comment:
I'm not sure it is necessary to add the ephemeral containers resource request, as the [KEP](https://github.com/kubernetes/enhancements/blob/master/keps/sig-node/20190212-ephemeral-containers.md) mentioned resources field is disallowed in ephemeral containers:

> [Some fields of v1.Container imply consequences for the entire pod. For example, one would expect setting resources to increase resources allocated to the pod, but this is not yet supported. We will disallow resources in Ephemeral Containers.](https://github.com/kubernetes/enhancements/blob/master/keps/sig-node/20190212-ephemeral-containers.md#updating-a-pod)
>
> [There are no guaranteed resources for ad-hoc troubleshooting. If troubleshooting causes a pod to exceed its resource limit it may be evicted.](https://github.com/kubernetes/enhancements/blob/master/keps/sig-node/20190212-ephemeral-containers.md#implementation-detailsnotesconstraints)

Reply:
+1 to what @draveness said. also from the KEP: "Ephemeral Containers lack guarantees for resources or execution"